File: 27.移除元素/27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        left=0
        right=len(nums)
        while left<right:
            if nums[left]==val:
                nums[left]=nums[right-1]
                right-=1
            else :left+=1 #不能无条件加，因为复制过来的尾部元素也有可能是和val相等的元素
        return left
        
                
            

File: 27.移除元素/27.cpp
class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int left = 0;
        for (int x : nums)
        {
            if (x != val)
            {
                nums[left] = x;
                left++;
            }
        }
        return left;
    }
};